Project/Src/Gui/TrackableVScrollBar.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System.Windows.Forms;
+
+namespace ICSharpCode.TextEditor;
+
+/// <summary>
+/// A vertical scrollbar control that tracks mouse down events to detect when the user is actively dragging the scrollbar.
+/// </summary>
+public class TrackableVScrollBar : VScrollBar
+{
+    const int WM_LBUTTONDOWN = 0x0201;
+
+    /// <summary>
+    /// Gets a value indicating whether the mouse button is currently pressed down on the scrollbar.
+    /// </summary>
+    public bool IsMouseDown { get; private set; }
+
+    public TrackableVScrollBar()
+    {
+        this.Scroll += VScrollBar_Scroll;
+    }
+
+    /// <summary>
+    /// Occurs when the user finishes scrolling and releases the mouse button.
+    /// </summary>
+    public event ScrollEventHandler MouseScrollEnded;
+
+    private void VScrollBar_Scroll(object sender, ScrollEventArgs e)
+    {
+        if (e.Type == ScrollEventType.EndScroll)
+        {
+            if (IsMouseDown)
+            {
+                IsMouseDown = false;
+
+                MouseScrollEnded?.Invoke(this, e);
+            }
+        }
+    }
+
+    protected override void WndProc(ref Message m)
```

Comment:
```suggestion
    protected override void OnMouseDown(MouseEventArgs e)
```

Reply:
Have you tried that? For me `OnMouseDown` doesn't fire which is why I had to resort to `WndProc`